sys/net/gnrc/routing/rpl/gnrc_rpl_auto_init.c

```diff
@@ -28,35 +28,38 @@
 
 void auto_init_gnrc_rpl(void)
 {
-#if (GNRC_NETIF_NUMOF == 1)
-    gnrc_netif_t *netif = gnrc_netif_iter(NULL);
-    if (netif == NULL) {
-        /* XXX this is just a work-around ideally this would happen with
-         * an `up` event of the interface */
-        LOG_INFO("Unable to auto-initialize RPL. No interfaces found.\n");
+    if (gnrc_netif_highlander()) {
+        gnrc_netif_t *netif = gnrc_netif_iter(NULL);
+        if (netif == NULL) {
+            /* XXX this is just a work-around ideally this would happen with
+             * an `up` event of the interface */
+            LOG_INFO("Unable to auto-initialize RPL. No interfaces found.\n");
+            return;
+        }
+        DEBUG("auto_init_gnrc_rpl: initializing RPL on interface %" PRIkernel_pid "\n",
+              netif->pid);
+        gnrc_rpl_init(netif->pid);
         return;
     }
-    DEBUG("auto_init_gnrc_rpl: initializing RPL on interface %" PRIkernel_pid "\n",
-          netif->pid);
-    gnrc_rpl_init(netif->pid);
-    return;
-#elif defined(GNRC_RPL_DEFAULT_NETIF)
-    if (gnrc_netif_get_by_pid(GNRC_RPL_DEFAULT_NETIF) != NULL) {
-        DEBUG("auto_init_gnrc_rpl: initializing RPL on interface %" PRIkernel_pid "\n",
-              GNRC_RPL_DEFAULT_NETIF);
-        gnrc_rpl_init(GNRC_RPL_DEFAULT_NETIF);
+    else if (GNRC_RPL_DEFAULT_NETIF) {
```

Comment:
Murdock does not like when `GNRC_RPL_DEFAULT_NETIF` is not defined. I fear this needs to stay `#ifdef`

```suggestion
    else {
#if defined(GNRC_RPL_DEFAULT_NETIF)
```

Reply:
Alternatively, you can (in a separate PR, please) make `GNRC_RPL_DEFAULT_NETIF` default to `KERNEL_PID_UNDEF` if not defined and check

```suggestion
    else if (GNRC_RPL_DEFAULT_NETIF != KERNEL_PID_UNDEF) {
```

Reply:
see https://github.com/RIOT-OS/RIOT/pull/13628